### PR TITLE
Bank Application Refactor

### PR DESCRIPTION
**Pull request by dmartin5470**
This PR redesigns the quickstart bank application and aims to make it easier and more straightforward to add new spec support in the future. As of right now, what's needed to implement a new spec endpoints is: 

* Add the necessary bank data in the `mount/bank/data.json` file and then adjust the `BankUserData` struct 
    An example of the struct modification for the upcoming openbanking brazil stuff would be 
    ```
    type Accounts struct {
	OBUK []models.OBAccount6 `json:"obuk"`
	OBBR <data type> `json:"obbr"`
    }
    ```
    The bank storage was refactored to not care about specific data (accounts, balances, etc), so actual processing and handling of this data is the responsibility of the endpoint logic interface implementor. 
* Add a new EndpointLogic implementation (e.g. `obbr_get_transactions_handler.go` that contains the implementor)
* Assign the new EndpointLogic implementation to the server and then set it up under the correct path

There was also one more change, and that was adding the `SPEC` configuration variable to .env, so that you can globally change what specification mode the quickstart project runs in.

### CODE REVIEW

File: apps/bank/errors.go [hunk 1]
```diff
@@ -1,11 +1,26 @@
 package main
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errNotFound            ErrNotFound
```
Comment (mbilski) [inline apps/bank/errors.go:9]: why do you need private global variable like this?

File: apps/bank/errors.go [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+	errAlreadyExists       ErrAlreadyExists
+	errForbidden           ErrForbidden
+	errInternalServer      ErrInternalServer
+	errUnprocessableEntity ErrUnprocessableEntity
+)
 
 type ErrNotFound struct {
 	resourceName string
```
Comment (mbilski) [inline apps/bank/errors.go:17]: resourceName should be public

File: apps/bank/errors.go [hunk 3]
```diff
@@ -14,6 +29,46 @@ type ErrAlreadyExists struct {
...
+func NewErrAlreadyExists(name string) ErrAlreadyExists {
+	return ErrAlreadyExists{name}
+}
+
 func (e ErrAlreadyExists) Error() string {
 	return fmt.Sprintf("resource %s already exists", e.resourceName)
 }
+
+type ErrForbidden struct {
+	err error
```
Comment (mbilski) [inline apps/bank/errors.go:41]: all the Error fields should be public

File: apps/bank/obuk_utils.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	acpClient "github.com/cloudentity/acp-client-go/models"
+)
+
+func OBUKMapError(err error) (int, models.OBError1) {
+	msg := err.Error()
+	resp := models.OBError1{
+		Message: &msg,
+	}
+
+	if errors.As(err, &errNotFound) {
```
Comment (mbilski) [inline apps/bank/obuk_utils.go:23]: this is really complicated solution to map a http status code to an error. Maybe do sth like this instead

```
type Error struct {
  Code int `json:"code"`
  Message string `json:"message"`
  Error error `json:"-"`
}

func (e *Error) WithError(err error) *Error {
  return &Error{
    Code: e.Code,
    Message: e.Message,
    Err: errors.WithStack(err),
  }
}

func (e *Error) Unwrap() error {
  return e.Err
}

func NewError(err error) *Error {
  var e Error

  if errors.As(err, &e) {
    return e
  }

  return &Error{Code: http.InternalServerError, Message: "unknown error", Err: error}
}

var (
  ErrNotFound = &Error{
    Code: http.StatusNotFound,
    Message: "entity not found",
  }
)
```

look at https://github.com/cloudentity/go-tools/blob/master/httpx/error.go for more inspiration (but you don't have to copy everything 1:1)

also read https://blog.golang.org/go1.13-errors

**Review by mbilski — APPROVED** [review]

**Review by ikawalec — APPROVED** [review]

**Pull request merged**